version/version.go

```diff
@@ -33,6 +34,14 @@ func GetVersion() *VersionInfo {
 	}
 }
 
+func GetVaultBuildDate() (time.Time, error) {
+	buildDate, err := time.Parse(time.RFC3339, BuildDate)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("failed to parse build date based on RFC3339: %w", err)
+	}
+	return buildDate, nil
+}
```

Comment:
Per the PR description, this is a copy of the existing, ENT-only [`vault.GetVaultBuildDate`](https://github.com/hashicorp/vault-enterprise/blob/f89429b963daa84ac832ed4479d0043f60231c8d/vault/licensing_ent.go#L109-L115) helper but we're making it shared in this common `version` package now.